### PR TITLE
fix: generate brick.g.dart in dart 3 (#343 #344)

### PR DESCRIPTION
**Pull request by tshedor**

### CODE REVIEW

File: example_graphql/pubspec.yaml [hunk 1]
```diff
@@ -16,9 +16,14 @@ dependencies:
...
+dependency_overrides:
+  brick_sqlite_generators:
+    path: ../packages/brick_sqlite_generators
+
 dev_dependencies:
-  brick_offline_first_with_graphql_build:
-    path: ../packages/brick_offline_first_with_graphql_build
+  brick_offline_first_with_rest_build:
+    path: ../packages/brick_offline_first_with_rest_build
+
```
Comment (mateominato) [inline example_graphql/pubspec.yaml:26]: intentional or debug relic?

File: example_rest/pubspec.yaml [hunk 2]
```diff
@@ -14,6 +14,10 @@ dependencies:
     sdk: flutter
   cupertino_icons: ^1.0.2
   sqflite: any
+  
+dependency_overrides:
+  brick_sqlite_generators:
+    path: ../packages/brick_sqlite_generators
```
Comment (mateominato) [inline example_rest/pubspec.yaml:20]: same

File: example_rest/lib/brick/adapters/pizza_adapter.g.dart [hunk 3]
```diff
@@ -13,7 +13,7 @@ Future<Map<String, dynamic>> _$PizzaToRest(Pizza instance,
     {required RestProvider provider, OfflineFirstWithRestRepository? repository}) async {
   return {
     'id': instance.id,
-    'toppings': instance.toppings?.map((e) => e.toString().split('.').last).toList(),
+    'toppings': instance.toppings?.map((e) => e.name).toList(),
```
Comment (mateominato) [inline example_rest/lib/brick/adapters/pizza_adapter.g.dart:16]: isn't that nice

File: packages/brick_sqlite_generators/pubspec.yaml [hunk 4]
```diff
@@ -4,7 +4,7 @@ homepage: https://github.com/GetDutchie/brick/tree/main/packages/brick_sqlite_ge
 issue_tracker: https://github.com/GetDutchie/brick/issues
 repository: https://github.com/GetDutchie/brick
 
-version: 3.0.1
+version: 3.0.2
```
Comment (mateominato) [inline packages/brick_sqlite_generators/pubspec.yaml:7]: the rest of these jumped to `3.1.0`

**Review by mateominato — APPROVED** [review]

File: example_graphql/pubspec.yaml (hunk 1 above)
Comment (tshedor) [reply example_graphql/pubspec.yaml:26]: Great catch

File: example_rest/pubspec.yaml (hunk 2 above)
Comment (tshedor) [reply example_rest/pubspec.yaml:20]: This one is intentional

File: packages/brick_sqlite_generators/pubspec.yaml (hunk 4 above)
Comment (tshedor) [reply packages/brick_sqlite_generators/pubspec.yaml:7]: Yeah `sqlite_generators` isn't a significant change

**Pull request merged**